mleap-runtime/src/main/scala/ml/combust/mleap/bundle/ops/feature/MapEntrySelectorOp.scala

```diff
@@ -0,0 +1,49 @@
+package ml.combust.mleap.bundle.ops.feature
+
+import ml.combust.bundle.BundleContext
+import ml.combust.bundle.dsl.{Bundle, Model, Node, Value}
+import ml.combust.bundle.op.OpModel
+import ml.combust.mleap.bundle.ops.MleapOp
+import ml.combust.mleap.core.feature.MapEntrySelectorModel
+import ml.combust.mleap.core.types.BasicType
+import ml.combust.mleap.runtime.MleapContext
+import ml.combust.mleap.runtime.types.BundleTypeConverters.{mleapToBundleBasicType, bundleToMleapBasicType, bundleToMleapNodeShape}
+import ml.combust.mleap.runtime.transformer.feature.MapEntrySelector
+
+class MapEntrySelectorOp extends MleapOp[MapEntrySelector[_, _], MapEntrySelectorModel[_, _]]{
+  override val Model: OpModel[MleapContext, MapEntrySelectorModel[_, _]] = new OpModel[MleapContext, MapEntrySelectorModel[_, _]] {
+
+    override val klazz: Class[MapEntrySelectorModel[_, _]] = classOf[MapEntrySelectorModel[_, _]]
+    override def opName: String = Bundle.BuiltinOps.feature.map_entry_selector
+
+    override def store(model: Model, obj: MapEntrySelectorModel[_, _])(implicit context: BundleContext[MleapContext]): Model = {
+      val keyType = mleapToBundleBasicType(obj.inputSchema.getField("key").get.dataType.base)
+      val valueType = mleapToBundleBasicType(obj.outputSchema.getField("output").get.dataType.base)
+
+      model.withValue("key_type", Value.basicType(keyType))
+        .withValue("value_type", Value.basicType(valueType))
+        .withValue("default_value", Value.anyAsType(obj.defaultValue, valueType))
+    }
+
+    override def load(model: Model)(implicit context: BundleContext[MleapContext]): MapEntrySelectorModel[_, _] = {
+      val keyBasicType = bundleToMleapBasicType(model.value("key_type").getBasicType)
+      val valueBasicType = model.value("value_type").getBasicType
+      val defaultValue = model.value("default_value").getAnyFromType(valueBasicType)
+      (keyBasicType, bundleToMleapBasicType(valueBasicType)) match {
+        case (BasicType.String, BasicType.Double) => MapEntrySelectorModel[String, Double](defaultValue)
+        case (k, v) => throw new UnsupportedOperationException(s"Can not load bundle of types $k, $v")
+      }
```

Comment:
So I tried really hard to make the MapEntrySelector be fully generic and dynamically compute the types.  I got everything to be generic except the Ops.  And I'm not sure if it is possible to make this part generic.  something something scala reflection something something compile time...

But clearly we can/should support more type settings, and I'd like to hear which ones we think are valuable.

Reply:
I had written some code to translate all the types back and forth between spark, mleap, and scala which all worked without issue.

The problem is that the types need to exist at runtime, which scala doesn't do usually (types are just for the compiler).  You can add `TypeTag` to things for the type to stick around, but that only seems to work for templated types.  This is why I could make all the transformer and model classes work since they have templated generics.  But I don't think we can make the ops be templatized (i.e., `MapEntrySelector[K,V]` is not a thing we can do).  Thus the code we have.

I did find some super gross options where you like exec a process to instantiate the types but I think just having the ops be case matched is better.